docs/references/tanstack-start/custom-signup-signin-pages.mdx

```diff
@@ -0,0 +1,87 @@
+---
+title: Build your own sign-in and sign-up pages for your TanStack Start app with Clerk
+description: Learn how to add custom sign-in and sign-up pages to your TanStack Start app with Clerk's prebuilt components.
+---
+
+This guide shows you how to use the [`<SignIn />`](/docs/components/authentication/sign-in) and [`<SignUp />`](/docs/components/authentication/sign-up) components with the [TanStack Router Catch-All Routes](https://tanstack.com/router/latest/docs/framework/react/guide/code-based-routing#splat--catch-all-routes) in order to build custom sign-in and sign-up pages for your TanStack Start app.
+
+If Clerk's prebuilt components don't meet your specific needs or if you require more control over the logic, you can rebuild the existing Clerk flows using the Clerk API. For more information, check out the [custom flows](/docs/custom-flows/overview) guides.
+
+The functionality of the components are controlled by the instance settings you specify in your [Clerk Dashboard](https://dashboard.clerk.com).
+
+> [!NOTE]
+> Just getting started with Clerk and TanStack Start? Check out the [quickstart tutorial](/docs/quickstarts/tanstack-start)!
+
+<Steps>
+  ## Build a sign-up page
+
+  The following example demonstrates how to render the [`<SignUp />`](/docs/components/authentication/sign-up) component.
+
+  ```tsx {{ filename: 'app/routes/sign-up.$.tsx' }}
+  import { SignUp } from '@clerk/tanstack-start'
+
+  export const Route = createFileRoute('/sign-up/$')({
+    component: Page,
+  })
+
+  function Page() {
+    return <SignUp />
+  }
+  ```
+
+  ## Build a sign-in page
+
+  The following example demonstrates how to render the [`<SignIn />`](/docs/components/authentication/sign-in) component.
+
+  ```tsx {{ filename: 'app/routes/sign-in.$.tsx' }}
+  import { SignIn } from '@clerk/tanstack-start'
+
+  export const Route = createFileRoute('/sign-in/$')({
+    component: Page,
+  })
+
+  function Page() {
+    return <SignIn />
+  }
+  ```
+
+  ## Configure your sign-up and sign-in pages
+
+  For SSR Mode, add environment variables for the `signIn`, `signUp`, `afterSignUp`, and `afterSignIn` paths:
+
+  ```env {{ filename: '.env' }}
+  CLERK_SIGN_IN_URL=/sign-in
+  CLERK_SIGN_UP_URL=/sign-up
+  CLERK_SIGN_IN_FALLBACK_URL=/
```

Comment:
`CLERK_SIGN_IN_FALLBACK_URL` should be `CLERK_SIGN_IN_FALLBACK_REDIRECT_URL` ?

Reply:
Hmm you are right, but seems that it's wrong in Remix's docs too ([ref](https://clerk.com/docs/references/remix/custom-signup-signin-pages#configure-your-sign-up-and-sign-in-pages))

Reply:
good point - i can update that